src/main/java/io/reactivex/internal/operators/maybe/MaybeFromFuture.java

```diff
@@ -50,17 +51,11 @@ protected void subscribeActual(MaybeObserver<? super T> observer) {
                 } else {
                     v = future.get(timeout, unit);
                 }
-            } catch (InterruptedException ex) {
-                if (!d.isDisposed()) {
-                    observer.onError(ex);
-                }
-                return;
-            } catch (ExecutionException ex) {
-                if (!d.isDisposed()) {
-                    observer.onError(ex.getCause());
+            } catch (Throwable ex) {
```

Comment:
Reduce the code duplication to one catch and extract the actual `Throwable` if `ex` was the `ExecutionException`.